pkg/apis/cluster/types.go

```diff
@@ -26,6 +26,19 @@ type Cluster struct {
 
 // ClusterSpec defines the desired state of a member cluster.
 type ClusterSpec struct {
+	// ID is the unique value for this cluster.
+	// It is typically generated during the progress of cluster registration:
+	// - generated by command 'karmadactl join' for clusters in Push mode.
+	// - generated by `karmada-agent` for clusters in Pull mode.
+	//
+	// It is not recommended to update this value unless you really know what you are doing, because
+	// it will/may be used to :
+	// - uniquely identify the clusters within the Karmada system.
+	// - compose the DNS name of multi-cluster services.
+	//
+	// +optional
+	ID string `json:"id,omitempty"`
```

Comment:
What is the difference between metadata.uid and cluster id?

Reply:
Updated.